sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -3,479 +3,624 @@
-    deadletterReason: string;
+  deadLetterErrorDescription: string;
+  deadletterReason: string;
 }
 
-export { delay }
+export { delay };
+
+export { Delivery };
 
-export { Delivery }
+// @public
+export type EntityStatus =
+  | "Active"
+  | "Creating"
+  | "Deleting"
+  | "ReceiveDisabled"
+  | "SendDisabled"
+  | "Disabled"
+  | "Renaming"
+  | "Restoring"
+  | "Unknown";
 
 // @public
-export type EntityStatus = "Active" | "Creating" | "Deleting" | "ReceiveDisabled" | "SendDisabled" | "Disabled" | "Renaming" | "Restoring" | "Unknown";
+export interface GetMessageIteratorOptions extends OperationOptions {}
 
-export { HttpOperationResponse }
+export { HttpOperationResponse };
+
+// @public
+export interface MessageAndContext<ContextT> {
+  context: ContextT;
+  message: ReceivedMessage;
+}
 
 // @public
 export type MessageCountDetails = {
-    activeMessageCount: number;
-    deadLetterMessageCount: number;
-    scheduledMessageCount: number;
-    transferMessageCount: number;
-    transferDeadLetterMessageCount: number;
+  activeMessageCount: number;
+  deadLetterMessageCount: number;
+  scheduledMessageCount: number;
+  transferMessageCount: number;
+  transferDeadLetterMessageCount: number;
 };
 
 // @public
 export interface MessageHandlerOptions {
-    autoComplete?: boolean;
-    maxConcurrentCalls?: number;
-    maxMessageAutoRenewLockDurationInSeconds?: number;
+  autoComplete?: boolean;
+  maxConcurrentCalls?: number;
+  maxMessageAutoRenewLockDurationInSeconds?: number;
 }
 
-export { MessagingError }
+// @public
+export interface MessageHandlers<ContextT> {
+  processError(err: Error): Promise<void>;
+  processMessage(message: ReceivedMessage, context: ContextT): Promise<void>;
+}
 
 // @public
-export interface OnError {
-    (error: MessagingError | Error): void;
+export type MessageIterator<ContextT> = AsyncIterable<MessageAndContext<ContextT>>;
+
+export { MessagingError };
+
+// @public
+export interface NonSessionReceiver<LockModeT extends "peekLock" | "receiveAndDelete"> {
+  // (undocumented)
+  close(): Promise<void>;
+  close(): Promise<void>;
+  diagnostics: {
+    peek(maxMessageCount?: number): Promise<ReceivedMessage[]>;
+    peekBySequenceNumber(
+      fromSequenceNumber: Long,
+      maxMessageCount?: number
+    ): Promise<ReceivedMessage[]>;
+  };
+  // (undocumented)
+  entityPath: string;
+  // (undocumented)
+  entityType: "queue" | "subscription";
+  // (undocumented)
+  getDeadLetterPath(): string;
+  getMessageIterator(options?: GetMessageIteratorOptions): MessageIterator<ContextType<LockModeT>>;
+  // (undocumented)
+  isReceivingMessages(): boolean;
+  receiveBatch(
+    maxMessages: number,
+    maxWaitTimeInSeconds?: number,
+    options?: ReceiveBatchOptions
+  ): Promise<{
+    messages: ReceivedMessage[];
+    context: ContextType<LockModeT>;
+  }>;
+  // (undocumented)
+  receiveDeferredMessage(sequenceNumber: Long): Promise<ServiceBusMessage | undefined>;
```

Comment:
`ServiceBusMessage` to `ReceivedMessage`.

Reply:
Issue filed: #7690